packages/playwright-core/src/web/traceViewer/sw.ts

```diff
@@ -37,11 +37,18 @@ async function loadTrace(trace: string, clientId: string, progress: (done: numbe
   if (entry)
     return entry.traceModel;
   const traceModel = new TraceModel();
-  let url = trace.startsWith('http') || trace.startsWith('blob') ? trace : `file?path=${trace}`;
+  const fetchFromInbuiltTraceViewerServer = !trace.startsWith('http') && !trace.startsWith('blob');
+  let url = fetchFromInbuiltTraceViewerServer ? `file?path=${trace}` : trace;
   // Dropbox does not support cors.
   if (url.startsWith('https://www.dropbox.com/'))
     url = 'https://dl.dropboxusercontent.com/' + url.substring('https://www.dropbox.com/'.length);
-  await traceModel.load(url, progress);
+  try {
+    await traceModel.load(url, progress);
+  } catch (error) {
+    if (fetchFromInbuiltTraceViewerServer && error instanceof Error && error.message === 'HTTP error Not Found')
```

Comment:
since local file URLs also get loaded over HTTP, check if its done via the internal trace viewer web-server and rewrite it that the local file could not be found (instead of showing the http error).